shoop/core/models/_service_behavior.py

```diff
@@ -168,3 +172,39 @@ def get_unavailability_reasons(self, service, source):
         groups_to_match = set(self.groups.all().values_list("pk", flat=True))
         if not bool(customer_groups & groups_to_match):
             yield ValidationError(_("Service is not available for any of the customers groups."))
+
+
+class RoundingBehavior(Enum):
+    ROUND_UP = 1
+    ROUND_DOWN = 2
+
+    class Labels:
+        ROUND_UP = _("round up")
+        ROUND_DOWN = _("round down")
+
+
+class RoundingBehaviorComponent(ServiceBehaviorComponent):
+    name = _("Rounding")
+    help_text = _("Round total order price to the nearest quant.")
+
+    rounding_quant = models.DecimalField(
+        max_digits=36, decimal_places=9,
+        verbose_name=_("rounding quant"))
+    rounding_behavior = EnumIntegerField(
+        RoundingBehavior,
+        default=RoundingBehavior.ROUND_DOWN,
+        verbose_name=_("rounding behavior"))
+
+    def get_costs(self, service, source):
+        total_price = source.total_price_of_products.value
+        rounded_amt = 0
+        if self.rounding_behavior == RoundingBehavior.ROUND_UP:
+            rounded_amt = nickel_round(total_price, quant=self.rounding_quant, rounding=decimal.ROUND_UP)
+        else:
+            rounded_amt = nickel_round(total_price, quant=self.rounding_quant, rounding=decimal.ROUND_DOWN)
+        rounded_amt = source.create_price(decimal.Decimal(rounded_amt - total_price))
+        yield ServiceCost(rounded_amt, self.rounding_behavior.label)
```

Comment:
1. You're assigning three different values to the `rounded_amt` variable. That makes it little hard to follow. Use different variables for different things.
2. You can operate with prices, no need to drop to bare decimals. `nickel_round(x, quant)` returns a `Price` if `x` is a `Price` and `quant` can still be bare `Decimal`.
3. Are these rounding modes "correct"? At least in Finnish stores prices are rounded with ROUND_HALF_UP method, e.g. 2,12 € will be rounded to 2,10 € and 2,13 € to 2,15 €. Neither ROUND_UP or ROUND_DOWN does that. We should provide  at least ROUND_HALF_UP too, but wouldn't hurt to provide some others too as long as they're supported by the decimal module.


I think this could be as simple as this:
```python
def get_costs(self, service source):
    total_price = source.total_price_of_products
    rounded = nickel_round(total_price, self.quant, self.mode)
    remainder = rounded - total_price
    yield ServiceCost(remainder, self.mode.label)
```
I'm assuming that the enum values for rounding mode use directly values of `decimal.ROUND_UP` etc. to avoid mapping them.